lib/client/sse_client.dart

```diff
@@ -104,7 +104,12 @@ class SseClient extends StreamChannelMixin<String> {
     } on ArgumentError catch (e) {
       _logger.warning('Invalid argument: $e');
     }
-    await HttpRequest.request('$_serverUrl&messageId=${++_lastMessageId}',
-        method: 'POST', sendData: encodedMessage, withCredentials: true);
+    try {
+      await HttpRequest.request('$_serverUrl&messageId=${++_lastMessageId}',
+          method: 'POST', sendData: encodedMessage, withCredentials: true);
+    } catch (_) {
+      _lastMessageId--;
```

Comment:
i'm not sure this is right. couldn't there have been some intermediate requests that incremented the messageId as well?

Reply:
Discussed offline. We'll log a severe message and close the connection to prevent unexpected hangs.